app/src/main/java/com/g/tragosapp/ui/cocktaildetails/CocktailsDetailFragment.kt

```diff
@@ -46,11 +52,64 @@ class CocktailsDetailFragment : Fragment() {
         binding.cocktailTitle.text = cocktail.name
         binding.cocktailDesc.text = cocktail.description
 
-        binding.btnSaveCocktail.setOnClickListener {
-            viewModel.saveCocktail(FavoritesEntity(cocktail.cocktailId,cocktail.image,cocktail.name,cocktail.description,cocktail.hasAlcohol))
-            Toast.makeText(requireContext(), "Cocktail saved to favorites", Toast.LENGTH_SHORT)
-                .show()
-        }
+        viewModel.getFavoriteCocktails().observe(viewLifecycleOwner, Observer { result ->
+            when (result) {
+                is Resource.Loading -> {
+                }
+                is Resource.Success -> {
+                    when (result.data.contains(cocktail)) {
+                        true -> {
+                            binding.btnSaveCocktail.setOnClickListener {
+                                viewModel.deleteCocktail(
```

Comment:
Why we are deleting a cocktail from the saveCocktail button ?

Reply:
BECAUSE THIS BUTTON IS BOTH ADD AND DELETE! you are asking very stupid questions. I added new functionality to you and you don't like it yet

Reply:
I did not asked stupid questions, I'm just asking nicely about the code, no need to bee rude

Reply:
> BECAUSE THIS BUTTON IS BOTH ADD AND DELETE! you are asking very stupid questions. I added new functionality to you and you don't like it yet

Then why is the ID called `btnSaveOrDeleteCocktail`? :thinking: it's not a stupid question, it's a completely valid question. 

The code and the variable name are mismatched, and this should either be fixed, or should be two separate buttons.